### PR TITLE
chore(ci): align docs workflow

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

As a follow-up to #3199:
- make the job an initial step of creating a PR for the release
- add a version as an input and use it to name the generated reference accordingly

Related to: https://github.com/Kong/kubernetes-ingress-controller/pull/3294

**Which issue this PR fixes**:

Part of https://github.com/Kong/team-k8s/issues/267.

### CODE REVIEW

**Comment by rainest** [comment]
Is there a way to dry-run this in forks first to confirm functionality and feel? May be a bit annoying with the multiple repos involved, but it should be possible to set up a personal KIC and docs fork and demo the release docgen/PR process with that.

**Comment by czeslavo** [comment]
> Is there a way to dry-run this in forks first to confirm functionality and feel? May be a bit annoying with the multiple repos involved, but it should be possible to set up a personal KIC and docs fork and demo the release docgen/PR process with that.

Here's a PR that was created by running `gh workflow run release_docs.yaml --ref chore/release-docs-workflow -f tag=v2.8.0`: https://github.com/Kong/docs.konghq.com/pull/4950

File: .github/ISSUE_TEMPLATE/release.yaml [hunk 1]
```diff
@@ -64,7 +64,7 @@ body:
   attributes:
     label: "**For major/minor releases only** Update Release documents"
     options:
-      - label: Trigger [synchronize_docs](https://github.com/Kong/kubernetes-ingress-controller/blob/main/.github/workflows/synchronize_docs.yaml) workflow.
+      - label: Trigger [release_docs](https://github.com/Kong/kubernetes-ingress-controller/blob/main/.github/workflows/release_docs.yaml) workflow.
```
Comment (rainest) [inline .github/ISSUE_TEMPLATE/release.yaml:67]:
```suggestion
      - label: Trigger [release_docs](https://github.com/Kong/kubernetes-ingress-controller/blob/main/.github/workflows/release_docs.yaml) workflow. Note that you will need to update the new version's navigation manifest (e.g. [for 2.7](https://github.com/Kong/docs.konghq.com/blob/main/app/_data/docs_nav_kic_2.7.x.yml) to use the new file after.
```

**Review by rainest — CHANGES_REQUESTED** [review]
Alright, looks good, just one minor addition since it doesn't (and probably can't) update the nav manifest to point to it.

**Review by rainest — APPROVED** [review]

**Pull request merged**